ingeniamotion/pdo.py

```diff
@@ -0,0 +1,337 @@
+        if isinstance(pdo_map_item, TPDOMapItem) and not isinstance(pdo_map, TPDOMap):
+            raise ValueError("Cannot add a TPDOItem to a RPDOMap")
+        pdo_map.add_item(pdo_map_item)
+
+    @staticmethod
+    def create_empty_rpdo_map() -> RPDOMap:
+        """
+        Create an empty RPDOMap.
+
+        Returns:
+            The empty RPDOMap.
+
+        """
+        return RPDOMap()
+
+    @staticmethod
+    def create_empty_tpdo_map() -> TPDOMap:
+        """
+        Create an empty TPDOMap.
+
+        Returns:
+            The empty TPDOMap.
+
+        """
+        return TPDOMap()
+
+    def set_pdo_maps_to_slave(
+        self,
+        rpdo_maps: Union[RPDOMap, List[RPDOMap]],
+        tpdo_maps: Union[TPDOMap, List[TPDOMap]],
+        servo: str = DEFAULT_SERVO,
+    ) -> None:
+        """
+        Map the PDOMaps to the slave.
+
+        Args:
+            rpdo_maps: The RPDOMaps to be mapped.
+            tpdo_maps: he TPDOMaps to be mapped.
+            servo: servo alias to reference it. ``default`` by default.
+
+        Raises:
+            ValueError: If there is a type mismatch retrieving the drive object.
+            ValueError: If not all instances of a RPDOMap are in the RPDOMaps to be mapped.
+            ValueError: If not all instances of a TPDOMap are in the TPDOMaps to be mapped.
+
+        """
+        drive = self.mc._get_drive(servo)
+        if not isinstance(drive, EthercatServo):
+            raise ValueError(f"Expected an EthercatServo. Got {type(drive)}")
+        if not isinstance(rpdo_maps, list):
+            rpdo_maps = [rpdo_maps]
+        if not isinstance(tpdo_maps, list):
+            tpdo_maps = [tpdo_maps]
+        if not all(isinstance(rpdo_map, RPDOMap) for rpdo_map in rpdo_maps):
+            raise ValueError("Not all elements of the RPDO map list are instances of a RPDO map")
+        if not all(isinstance(tpdo_map, TPDOMap) for tpdo_map in tpdo_maps):
+            raise ValueError("Not all elements of the TPDO map list are instances of a TPDO map")
+        drive.set_pdo_map_to_slave(rpdo_maps, tpdo_maps)
+
+    def clear_pdo_mapping(self, servo: str = DEFAULT_SERVO) -> None:
+        """
+        Clear the PDO mapping within the servo.
+
+        Args:
+            servo: servo alias to reference it. ``default`` by default.
+
+        Raises:
+            ValueError: If there is a type mismatch retrieving the drive object.
+
+        """
+        drive = self.mc._get_drive(servo)
+        if not isinstance(drive, EthercatServo):
+            raise ValueError(f"Expected an EthercatServo. Got {type(drive)}")
+        drive.reset_rpdo_mapping()
+        drive.reset_tpdo_mapping()
+
+    def start_pdos(
+        self,
+        interface_name: str,
+        refresh_rate: Optional[float] = None,
+    ) -> None:
+        """
+        Start the PDO exchange process.
+
+        Args:
+            interface_name: The interface name where the slaves are connected to.
```

Comment:
Working with the interface name is a pain, maybe we must make a decision and only allow one EtherCAT network. Then you always will know the Network.

Reply:
But that will have a problem, how will we work when CANopen supports PDOs? it must be defined.

Reply:
Done